src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/SlimefunMachine.java

```diff
@@ -80,5 +99,34 @@ public MultiBlock toMultiBlock() {
 	public Iterator<ItemStack[]> recipeIterator() {
 		return this.recipes.iterator();
 	}
-
+	
+	// Overloaded method for finding a potential output chest. Fallbacks to the old system of putting the adding back into the dispenser.
+	// Optional last argument Inventory placeCheckerInv is for multiblock machines that create a dummy inventory to check if there's a space for the adding,
+	// i.e. Enhanced crafting table
+	public static Inventory findValidOutputInv(ItemStack adding, Block dispBlock, Inventory dispInv) {
+		return findValidOutputInv(adding, dispBlock, dispInv, dispInv);
+	}
+	
+	public static Inventory findValidOutputInv(ItemStack product, Block dispBlock, Inventory dispInv, Inventory placeCheckerInv) {
+		Inventory outputInv = null;
+		for (BlockFace face : faces) {
+			Block potentialOutput = dispBlock.getRelative(face);
+			if (BlockStorage.hasBlockInfo(potentialOutput) && BlockStorage.checkID(potentialOutput).equals("OUTPUT_CHEST")) {
+				// Found the output chest! Now, let's check if we can fit the adding in it.
+				Inventory chestInv = ((Chest) potentialOutput.getState()).getInventory();
```

Comment:
Nitpicky, but instead of casting to Chest, use a super-interface (I think it's called Container?) 
Just in case someone adds an Output Shulker Box or so? IDK XD

Reply:
Sure ^^